x-pack/platform/plugins/shared/observability_ai_assistant/server/utils/rewrite_query.ts

```diff
@@ -0,0 +1,235 @@
+  });
+
+  const parts: string[] = [
+    `Below is a conversation from the user. Given
+      the information in the conversation, write queries that
+      can be executed against the user's knowledge base.
+
+      A semantic query is often a question, but it can be any
+      text that will be used for semantic searches.
+
+      Use keyword searches for named entities.
+
+      You can boost a query. A higher value for \`boost\` means
+      that a document that matches it will be scored higher.
+      
+      ### Example Prompt-to-Queries Rewrites
+
+      **Example 1**
+
+      User Prompt:  
+      "What can you tell me about my open alerts?"
+
+      Rewrite into Queries:  
+      - Semantic: "information on open alerts"  
+        Keywords: []  
+        Boost: 1  
+
+      - Semantic: "details and resolution steps for open alerts"  
+        Keywords: []  
+        Boost: 2  
+
+      ---
+
+      **Example 2**
+
+      User Prompt:  
+      "Visualize the count of my zookeeper logs over time."
+
+      Context:  
+      "A snippet in GitHub mentions 'zookeeper staging'."
+
+      Rewrite into Queries:  
+      - Semantic: "zookeeper logs count over time"  
+        Keywords: ["zookeeper"]  
+        Boost: 2  
+
+      - Semantic: "zookeeper staging logs reference"  
+        Keywords: ["zookeeper staging"]  
+        Boost: 1  
+
+      ---
+
+      **Example 3**
+
+      User Prompt:  
+      "Who is on call for the PaymentService right now?"
+
+      Context:  
+      "JIRA issues sometimes refer to 'payment-service' (with a hyphen)."
+
+      Rewrite into Queries:  
+      - Semantic: "on call schedule for PaymentService"  
+        Keywords: ["PaymentService", "payment-service"]  
+        Boost: 2  
+
+      - Semantic: "who is currently on call"  
+        Keywords: []  
+        Boost: 1  
+
+      ---
+
+      **Example 4**
+
+      User Prompt:  
+      "Any logs containing the IP 192.168.1.10?"
+
+      Context:  
+      "There is also a reference in Confluence to user 'alice' with that IP."
+
+      Rewrite into Queries:  
+      - Semantic: "IP 192.168.1.10 in logs"  
+        Keywords: ["192.168.1.10"]  
+        Boost: 2  
+
+      - Semantic: "logs referencing user alice"  
+        Keywords: ["alice"]  
+        Boost: 1  
+
+      ---
+
+      **Example 5**
+
+      User Prompt:  
+      "Where can I find docs for ephemeral Elasticsearch cluster setup?"
+
+      Context:  
+      "Confluence references ephemeral clusters with keyword 'ephemeral-cluster'."
+
+      Rewrite into Queries:  
+      - Semantic: "ephemeral Elasticsearch cluster setup docs"  
+        Keywords: ["ephemeral-cluster"]  
+        Boost: 2  
+
+      - Semantic: "documentation for ephemeral cluster configuration"  
+        Keywords: []  
+        Boost: 1`,
+  ];
+
+  if (lastUserMessage) {
+    parts.push(`Pay special attention to the last user message.
```

Comment:
```suggestion
    parts.push(`Pay special attention to the user prompt. This is the last user message in the conversation history and reflects the focus of the user's inquiry.
```

Reply:
This got me thinking: should we also pay special attention to the original user message and constantly keep this in mind? So every new user message should be seen in relation to the original question? 
I know that some conversation can diverge greatly, so might not always work great.